WebApplication/Services/MigrationBucketKeyProvider.cs

```diff
@@ -1,14 +1,44 @@
 ﻿using System.Threading.Tasks;
+using WebApplication.Utilities;
 
 namespace WebApplication.Services
 {
     public class MigrationBucketKeyProvider : IBucketKeyProvider
     {
-        public string BucketKey { private get; set; }
+        private readonly BucketPrefixProvider _bucketPrefixProvider;
+        private readonly ResourceProvider _resourceProvider;
+        private string BucketKey = "";
+        public string AnonymousBucketKey {get;}
+
+        public MigrationBucketKeyProvider(BucketPrefixProvider bucketPrefixProvider, ResourceProvider resourceProvider)
+        {
+            _bucketPrefixProvider = bucketPrefixProvider;
+            _resourceProvider = resourceProvider;
+            AnonymousBucketKey = resourceProvider.BucketKey;
+        }
 
         public Task<string> GetBucketKeyAsync()
         {
             return Task.FromResult(BucketKey);
         }
+        public string SetBucketKeyFromOld(string bucketKeyOld)
+        {
+            string [] splittedBucketKeyOld = bucketKeyOld.Split('-');
+
+            if (splittedBucketKeyOld[0] == "projects")
```

Comment:
I think we should move this to some shared property. It is the same as projects mentioned in ResourceProvider class